third_party/parser/cc/lexer.rl

```diff
@@ -2131,6 +2131,23 @@ void lexer::set_state_expr_value() {
         fnext expr_end; fbreak;
       };
 
+      ':' ( '@'  %{ tm = p - 1; }
+          | '@@' %{ tm = p - 2; }
+          ) [0-9]*
+      => {
+        if (version == ruby_version::RUBY_27) {
```

Comment:
Interesting, this is the first time I'm learning that we have some sort of multiple version support in the parser.

We always invoke the driver with `ruby_version::RUBY_27`? Seems kind of odd that we go to the effort of maintaining that if we're always invoking it at a fixed version. `¯\_(ツ)_/¯`.

Reply:
That was my concern too: https://github.com/sorbet/sorbet/pull/3226#discussion_r444233892

We maintain a lot of code that is never executed. 

That said, with all this it would be quite easy to add a `--ruby-version` to Sorbet so it behave properly on a fixed version of the syntax.

Reply:
It might be easy to make the parser changes, but I don't think it would be easy to make it also support different versions of RBIs and different semantics in the code.